drivers/sensor/bmi270/bmi270_crt.c

```diff
@@ -0,0 +1,743 @@
+/*
+ * Copyright (c) 2023 Bosch Sensortec GmbH. All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/logging/log.h>
+
+#include "bmi270.h"
+#include "bmi270_config_file.h"
+LOG_MODULE_DECLARE(bmi270, CONFIG_SENSOR_LOG_LEVEL);
+
+#define BMI270_REG_FEATURES_0_END   0x3F
+#define BMI270_INTER_WRITE_DELAY_US 1000
+
+#define BMI270_CRT_STATUS_CHECK_RETRIES    15
+#define BMI270_NVM_STATUS_CHECK_RETRIES    100
+#define BMI270_STATUS_CHECK_POLL_PERIOD_US 10000
+
+/** @name Enum to define status of gyroscope trigger G_TRIGGER */
+typedef enum {
```

Comment:
I'm not sure where Zephyr documents these things, but they don't use tyepdef declarations. Enumeration members should also be prepended with the enumeration name to prevent naming collisions

```
enum {
	BMI270_CRT_TRIGGER_STAT_SUCCESS = 0,
	...
};
```